internal/client/config.go

```diff
@@ -341,6 +370,30 @@ func (c *Config) ConfigAccess() (clientcmd.ConfigAccess, error) {
 	return c.clientConfig().ConfigAccess(), nil
 }
 
+func (c *Config) newInclusterContext() *api.Context {
+	ns, _, _ := c.clientConfig().Namespace()
+	if ns == "" {
+		ns = "default"
+	}
+	nc := api.NewContext()
+	nc.LocationOfOrigin = incluster
+	nc.Cluster = incluster
+	nc.Namespace = ns
+	nc.AuthInfo = incluster
+	return nc
+}
+
+func (c *Config) isIncluster(cfg api.Config) bool {
+	if (cfg.CurrentContext == "" || cfg.CurrentContext == incluster) &&
```

Comment:
Might as well return the bool expression. Also we could use a helper to check for empty i.e isEmpty(*string) bool

Reply:
Ok.